bundles/tools.vitruv.change.propagation/src/tools/vitruv/change/propagation/ChangePropagationMode.java

```diff
@@ -0,0 +1,22 @@
+package tools.vitruv.change.propagation;
+
+/**
+ * The mode in which changes are propagated by executing change propagation specifications.
+ */
+public enum ChangePropagationMode {
+	/**
+	 * Only executes transformations adjacent to the changed model's metamodel.
+	 */
+	SINGLE_STEP,
+
+	/**
+	 * Executes transformations transitively and cyclic until no further changes are produced.
+	 */
+	TRANSITIVE_CYCLIC,
+
+	/**
+	 * Executes transformations transitively, but does not further propagate changes to a leaf model (i.e., a model with only
+	 * one transformation from and to another model).
+	 */
+	TRANSITIVE_EXCEPT_LEAVES
```

Comment:
This seems like a pretty weird exception case. I guess we need it for some specific use case?
I would have expected to have something like _transitive but only touch each domain once_. This would be kind of similar to the leaves concept when considering domains as a tree structure, root being the domain where changes originate and children those with consistency relations to the parent.